library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcMetaDataUtility.java

```diff
@@ -0,0 +1,344 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.db.jdbc;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datatorrent.lib.util.KeyValPair;
+
+/**
+ * A utility class used to retrieve the metadata for a given unique key of a SQL
+ * table. This class would emit range queries based on a primary index given
+ * 
+ * @Input - dbName,tableName, primaryKey
+ * @Output - map<operatorId,prepared statement>
+ *
+ */
+public class JdbcMetaDataUtility
+{
+  private static String DB_DRIVER = "com.mysql.jdbc.Driver";
+  private static String DB_CONNECTION = "";
+  private static String DB_USER = "";
+  private static String DB_PASSWORD = "";
+  private static String TABLE_NAME = "";
+  private static String KEY_COLUMN = "";
+  private static String WHERE_CLAUSE = null;
+  private static String COLUMN_LIST = null;
+
+  private static Logger LOG = LoggerFactory.getLogger(JdbcMetaDataUtility.class);
+
+  public JdbcMetaDataUtility()
+  {
+
+  }
+
+  public JdbcMetaDataUtility(String dbConnection, String tableName, String key, String userName, String password)
+  {
+    DB_CONNECTION = dbConnection;
+    DB_USER = userName;
+    DB_PASSWORD = password;
+    TABLE_NAME = tableName;
+    KEY_COLUMN = key;
+  }
+
+  private static Connection getDBConnection()
+  {
+
+    Connection dbConnection = null;
+
+    try {
+      Class.forName(DB_DRIVER);
+    } catch (ClassNotFoundException e) {
+      LOG.error("Driver not found", e);
+    }
+
+    try {
+      dbConnection = DriverManager.getConnection(DB_CONNECTION, DB_USER, DB_PASSWORD);
+      return dbConnection;
+    } catch (SQLException e) {
+      LOG.error("Exception in getting connection handle", e);
+    }
+
+    return dbConnection;
+
+  }
+
+  private static String generateQueryString()
+  {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT COUNT(*) as RowCount from " + TABLE_NAME);
+
+    if (WHERE_CLAUSE != null) {
+      sb.append(" WHERE " + WHERE_CLAUSE);
+    }
+
+    return sb.toString();
+  }
+
+  /**
+   * Finds the total number of rows in the table
+   */
+  private static long getRecordRange(String query) throws SQLException
+  {
+    long rowCount = 0;
+    Connection dbConnection = null;
+    PreparedStatement preparedStatement = null;
+
+    try {
+      dbConnection = getDBConnection();
+      preparedStatement = dbConnection.prepareStatement(query);
+
+      ResultSet rs = preparedStatement.executeQuery();
+
+      while (rs.next()) {
+        rowCount = Long.parseLong(rs.getString("RowCount"));
```

Comment:
Suggest to use index instead of a column alias. Also seems like this is a simple count utility. Can you rename it to something like "getTupleCount()" ?